examples/stable_diffusion_v2/text_to_image.py

```diff
@@ -433,27 +451,34 @@ def main(args):
     )
     args = parser.parse_args()
 
-    # overwrite env var by parsed arg
+    # check args
     if args.version:
         os.environ["SD_VERSION"] = args.version
+        if args.version not in _version_cfg:
+            raise ValueError(f"Unknown version: {args.version}. Supported SD versions are: {list(_version_cfg.keys())}")
     if args.ckpt_path is None:
-        if args.version in ["1.5_cn", "wukong"]:
-            args.ckpt_path = "models/wukong-huahua-ms.ckpt"
-        elif args.version.startswith("1."):  # 1.x, 1.5
-            args.ckpt_path = "models/sd_v1.5-d0ab7146.ckpt"
-        elif args.version == "2.0-v768":
-            args.ckpt_path = "models/sd_v2_768_v-e12e3a9b.ckpt"
-        else:
-            args.ckpt_path = "models/sd_v2_base-57526ee4.ckpt"
+        ckpt_name = _version_cfg[args.version][0]
+        args.ckpt_path = "models/" + ckpt_name
+
+        desire_size = _version_cfg[args.version][2]
+        if args.H != desire_size or args.W != desire_size:
+            logger.warning(
+                f"The optimal H, W for SD {args.version} is ({desire_size}, {desire_size}) . But got ({args.H}, {args.W})."
+            )
+
+        # download if not exists or not complete
+        ckpt_incomplete = False
+        if os.path.exists(args.ckpt_path):
+            if os.path.getsize(args.ckpt_path) < _MIN_CKPT_SIZE:
+                ckpt_incomplete = True
+                print(
+                    f"WARNING: The checkpoint size is too small {args.ckpt_path}. Please check and remove it if it is incomplete!"
+                )
+        if not os.path.exists(args.ckpt_path):
+            print(f"Start downloading checkpoint {ckpt_name} ...")
+            download_checkpoint(_URL_PREFIX + "/" + ckpt_name, "models/")
     if args.config is None:
-        if args.version in ["1.5_cn", "wukong"]:
-            args.config = "configs/v1-inference-chinese.yaml"
-        elif args.version.startswith("1."):  # 1.x, 1.5
-            args.config = "configs/v1-inference.yaml"
-        elif args.version == "2.0-v768":
-            args.ckpt_path = "models/v2-vpred-inference.yaml"
-        else:
-            args.config = "configs/v2-inference.yaml"
+        args.config = "configs/" + _version_cfg[args.version][1]
```

Comment:
os.path.join?

Reply:
same. the folder name is fixed anyway.

Reply:
I think it's a backward slash \ on windows

Reply:
agree. fixed